### PR TITLE
Fixes #27801 - Add cancel button to the tasks table

### PR DESCRIPTION
**Pull request by MariaAga**
Adds a button that will cancel a task in the table (if the task can be canceled) and display a toast notification based on the answer
Based on: https://github.com/theforeman/foreman-tasks/pull/452

### CODE REVIEW

**Comment by sharvit** [comment]
Needs rebase

**Comment by sharvit** [comment]
The commit list doesn't look right :/

**Comment by MariaAga** [comment]
@sharvit Fixed and rebased thanks for noticing

File: webpack/ForemanTasks/Components/TaskDetails/Components/Task.js [hunk 1]
```diff
@@ -37,6 +38,8 @@ class Task extends Component {
       showForceUnlockModal,
       toggleUnlockModal,
       toggleForceUnlockModal,
+      cancelTask,
+      action,
```
Comment (sharvit) [inline webpack/ForemanTasks/Components/TaskDetails/Components/Task.js]: action needs to have a `propType`

Comment (MariaAga) [reply webpack/ForemanTasks/Components/TaskDetails/Components/Task.js]: they have a prop type in [Tasks Info](https://github.com/theforeman/foreman-tasks/blob/2c97ec1653c756cd580e4bedb5c4f040608cc63e/webpack/ForemanTasks/Components/TaskDetails/Components/TaskInfo.js#L217)  [Imported here](https://github.com/theforeman/foreman-tasks/blob/2c97ec1653c756cd580e4bedb5c4f040608cc63e/webpack/ForemanTasks/Components/TaskDetails/Components/Task.js#L168)
Should I move it from taskInfo to Task?

Comment (sharvit) [reply webpack/ForemanTasks/Components/TaskDetails/Components/Task.js]: Thanks! I missed that, looks good in this case :+1:

**Comment by adamruzicka** [comment]
Needs a rebase now that https://github.com/theforeman/foreman-tasks/pull/446 went in

**Comment by MariaAga** [comment]
Rebased.
@sharvit I added to the cancel action a line to refresh the dashboard as well as the table

File: webpack/ForemanTasks/Components/TasksTable/TasksTableActions.js [hunk 2]
```diff
@@ -1,7 +1,43 @@
...
 import { TASKS_TABLE_ID } from './TasksTableConstants';
 import { getApiPathname } from './TasksTableHelpers';
+import { fetchTasksSummary } from '../TasksDashboard/TasksDashboardActions';
 
 export const getTableItems = url =>
   getTableItemsAction(TASKS_TABLE_ID, getURIQuery(url), getApiPathname(url));
+
+export const cancelTaskAction = (id, name) => async dispatch => {
+  await dispatch(cancelTask(id, name));
+  const url = window.location.href;
```
Comment (sharvit) [inline webpack/ForemanTasks/Components/TasksTable/TasksTableActions.js]: Shouldn't it use the `history` object now?

**Comment by adamruzicka** [comment]
Well done @MariaAga , thank you. Also kudos to @sharvit for the review

**Pull request merged**